File: Lemur/src/frc/team4330/robot/commands/StartAgitator.java
package frc.team4330.robot.commands;

import edu.wpi.first.wpilibj.command.Command;
import frc.team4330.robot.Robot;

/**
 * A command that runs the agitator/feeder (independent of the shooter.)
 * 
 * @author Amanda
 */
public class StartAgitator extends Command {

	public StartAgitator() {
		requires(Robot.bamm);
	}
	
	@Override
	protected void initialize() {
		Robot.bamm.start();
	}
	
	@Override
	protected boolean isFinished() {
		return false;
	}
	
	@Override
	protected void end() {
		Robot.bamm.stop();
	}

}

File: Lemur/src/main/java/frc/team4330/robot/commands/StartAgitator.kt
package frc.team4330.robot.commands

import edu.wpi.first.wpilibj.command.Command
import frc.team4330.robot.Robot

/**
 * A command that runs the agitator/feeder (independent of the shooter.)
 *
 * @author Amanda
 */
class StartAgitator : Command() {
    init {
        requires(Robot.bamm)
    }

    override fun initialize() {
        Robot.bamm.start()
    }

    override fun isFinished(): Boolean {
        return false
    }

    override fun end() {
        Robot.bamm.stop()
    }

    override fun interrupted() {
        println(this.name + "was interrupted!")
        end()
    }
}
